source/extensions/filters/http/compressor/compressor_filter.h

```diff
@@ -160,6 +160,18 @@ class CompressorFilterConfig {
 };
 using CompressorFilterConfigSharedPtr = std::shared_ptr<CompressorFilterConfig>;
 
+class CompressorPerRouteFilterConfig : public Router::RouteSpecificFilterConfig {
```

Comment:
Should the filter config factory be changed to provide the `CompressorPerRoute` in overriding `FactoryBase`(https://github.com/envoyproxy/envoy/blob/65f000334aa025bac5763428086abcaf540d0312/source/extensions/filters/http/common/factory_base.h#L62)?

Reply:
The examples I emulated got the  per-route config by calling Http::Utility::resolveMostSpecificPerFilterConfig in decodeHeaders and/or encodeHeaders. I thought that the per-route config wasn't available in the filter config factory.

I'm calling Http::Utility::resolveMostSpecificPerFilterConfig in CompressorFilter::compressionEnabled. If there's a better way, can you point me at an example?

Reply:
I'm not saying what you did was wrong, but it seems incomplete IIUC how filters are configured.
For example, the [ext_proc filter](https://github.com/envoyproxy/envoy/blob/e4ae2259add5a327c4d67de18a82df5d78a801b4/source/extensions/filters/http/ext_proc/config.h#L16-L17) uses the factory that references both the filter config and the per-route config.

Reply:
Thank you for pointing me in the right direction. I've added the missing pieces, as well as integration tests to confirm.